### PR TITLE
Copper item repairing, mainlining, and tweaks

### PR DESCRIPTION
**Pull request by chaosvolt**
This implements the suggested idea of expanding soldering to allow repairing copper items, along with the suggestion of mainlining some of the copper items from More Survival Tools. Should probably also mainline those doors as @Rivet-the-Zombie suggested, though that should possibly be in a separate PR. @_@
- Added a scrap copper item.
- Added ability to repair copper items using scrap copper.
- Made deconstruction recipes for copper items return sensible amounts of small copper by weight.
- Changed ammo recipes to use copper scrap instead of copper tubes, and at more sensible ratios, relative to the required amounts of small copper.
- Changed copper recipes and items to more closely follow ye olde conservation of mass.
- Made some furniture yield a portion of their copper tubing as copper scrap instead.
- Mainlined copper pot, knife, spear, and axe from More Survival Tools, with updates to allow use of copper scrap for the recipes.
- Added copper scrap to a few item groups, supplementing and/or replacing copper tubing in various cases.

### CODE REVIEW

File: data/json/recipes/recipe_deconstruction.json [hunk 1]
```diff
@@ -125,7 +125,93 @@
...
+  "subcategory": "CSC_NONCRAFT",
+  "difficulty": 0,
+  "time": 1200,
+  "reversible": true,
+  "autolearn": false,
+  "qualities":[
+    {"id":"HAMMER","level":1,"amount":1}
+  ], "components": [
+    [
+      [ "stick", 1 ]
```
Comment (Coolthulhu) [inline data/json/recipes/recipe_deconstruction.json]: Should probably also allow 2-by-2. Sticks are supposed to be mostly just actual sticks.

File: data/json/recipes/recipe_deconstruction.json [hunk 2]
```diff
@@ -125,7 +125,93 @@
...
+  "category": "CC_NONCRAFT",
+  "subcategory": "CSC_NONCRAFT",
+  "difficulty": 0,
+  "time": 600,
+  "reversible": true,
+  "autolearn": false,
+  "qualities":[
+    {"id":"HAMMER","level":1,"amount":1}
+  ], "components": [
+    [
```
Comment (Coolthulhu) [inline data/json/recipes/recipe_deconstruction.json]: No bone/piece of wood/rag for handle?

Comment (Coolthulhu) [inline data/json/recipes/recipe_deconstruction.json]: Oh wait, those are just uncrafts.

No need to have separate uncrafts when you have crafting recipes for this stuff.

File: data/json/recipes/recipe_others.json [hunk 3]
```diff
@@ -3501,6 +3542,24 @@
...
+  "reversible": false,
+  "autolearn": true,
+  "qualities":[{"id":"HAMMER","level":1,"amount":1}],
+  "tools": [
+    [
+      [ "toolset", 10 ],
+      [ "fire", -1 ]
+    ]
+  ],
+  "components": [ [ [ "copper", 111 ] ] ]
```
Comment (Coolthulhu) [inline data/json/recipes/recipe_others.json]: Why not round to 100?

**Comment by chaosvolt** [comment]
Check the comment for the scrap_copper item in melee.json, there's a reason it's at 222 weight and 111 copper to make.

As for copper knife issues, if I recall I based off either the stone knife or the makeshift one...

**Comment by Coolthulhu** [comment]
If allowing reversing the recipe makes it require weird components (mold), it should be fixed by making those components unnecessary in `player::can_disassemble` in crafting.cpp.

**Comment by Coolthulhu** [comment]
> Check the comment for the scrap_copper item in melee.json, there's a reason it's at 222 weight and 111 copper to make.

Rounding it down to 100 wouldn't hurt "realism" much, while it would make it more convenient.
What is anyone going to do with 89 copper anyway?

Same would be good for other copper recipes - rounding them to nearest 100 or 50.

**Comment by chaosvolt** [comment]
Hmm. Actual reason for the separate dismantling recipes is because they might not need a fire (tool, not component) to dismantle, though I could make them reversible instead as fire isn't that hard to come by.

**Comment by Coolthulhu** [comment]
Adding a separate recipe for uncrafting is just asking for an "oops I forgot".

We probably should get rid of some of the existing ones, like one for bindles (so they no longer magically turn leather into rags and two-by-fours into sticks).

**Comment by chaosvolt** [comment]
Ah, maybe. I'll work on unfucking my copper recipes for the moment. Copper pot might still warrant a separate uncraft recipe. Though it might in theory be possible to abuse different melting points to melt off the tin lining first, before melting down the copper.

**Comment by Night-Pryanik** [comment]
After merging we should (maybe even me) extend the same processing to aluminum items.

**Comment by chaosvolt** [comment]
Ah, the whole conservation of mass thing?

**Comment by Night-Pryanik** [comment]
I mean make something like "aluminum scrap" and make aluminum items be repairable with this scrap.

**Comment by chaosvolt** [comment]
Ah right, I see. Not sure how beefy ingots are anyway, and I figured a smaller chunk would make more sense for copper, at least. ^^"

**Comment by kevingranade** [comment]
Aluminum is super hard to work with from what I understand, it would need a
different class of welder at a minimum, I'm sure @rivet can tell you all
about it.

**Comment by chaosvolt** [comment]
Nice to hear that long after @Night-Pryanik merged that "soldering with aluminum" thing. >.>

**Comment by Coolthulhu** [comment]
Reminds me: our melee weapons are badly categorized, they're spread across melee.json and tools.json with no particular pattern (except that most of them are wrongly put in the opposite category).

**Comment by chaosvolt** [comment]
last I checked, the critera was having a use_action. Some items in melee.json that now have a use action presumably didn't when they were first created.

**Comment by Night-Pryanik** [comment]
Soldering aluminum isn't that hard tool-wise. The main difficulty is an aluminum oxide film, which is a chemistry matter, and makes soldering a bit arduous, but not impossible. It doesn't need super powerful soldering iron or welder.

**Comment by Rivet-the-Zombie** [comment]
>it would need a different class of welder at a minimum

That kinda depends on particulars: you can easily TIG weld with aluminum filler rods, GMAW (MIG) with a proper wire-feed gun and aluminum-specific shield gas, and even SMAW (stick) with the right sort of electrodes/polarity.

This would mean we'd have to start differentiating between the various types of welders, and we'd have to start including various types of shield gas, feed wire, SMAW electrodes, TIG electrodes, etc. As it stands, our welding system is simple enough that a new player doesn't have to attend night classes at their local technical school to learn how it works, and I'm okay with that. Complexity and realism is great until it starts to cut into the fun, and I'm going to guess that most of our playerbase isn't here to study to become engineers, but to kill zombies.

**Comment by chaosvolt** [comment]
Basically, yeah. Might be too much complexity for little benefit.

**Comment by Coolthulhu** [comment]
@chaosvolt Did you test this PR WITHOUT having the mod active or only with? It complains about ingredients that aren't yet mainlined.

**Comment by chaosvolt** [comment]
Um, crap. Let me check, I might've failed to mainline a few things. @_@

**Comment by Rivet-the-Zombie** [comment]
Mainline all the things!

Seriously though, @chaosvolt, I like your additions and would love to see about 75% of the stuff you've got in your various mods moved in the base game - most everything you've added that doesn't directly contradict the base canon - most of the survival stuff immediately springs to mind as particularly good candidates for mainlining.

**Comment by Coolthulhu** [comment]
It's just the resin cords, it seems.
For now you could just keep the recipe override in the mod and delete the cords from mainlined recipes.

Resin cords would make OK mainline material under one circumstance: we also allow duct tape to have their use function (which would make sense).

**Comment by chaosvolt** [comment]
As I figured, resinous cord should maybe be tweaked some before being mainlined. Also @Rivet-the-Zombie I still need to do that with the doors. >.>

**Comment by Rivet-the-Zombie** [comment]
>Also @Rivet-the-Zombie I still need to do that with the doors.

I'm on tenterhooks here; PR and I will merge it.

**Comment by chaosvolt** [comment]
Doh. Wondering why I was seeing old changes before realizing I was in my Cataclysm folder, not the Github folder. >_>

**Comment by Coolthulhu** [comment]
Merged as-is, but I'm not sure if there shouldn't be more requirements for stuff.
Can copped be worked just by bonfire temperatures?
Shouldn't it require some tools, like tongs, pots or something like that?

**Comment by chaosvolt** [comment]
Hmm, might be. It seems possibly okay for the cruder tools like the knife, but the copper pot might need more complexity.

Most likely I blame having started with the usual primitive tool recipes and edited from there. ^^"

**Comment by chaosvolt** [comment]
And now...sorry for the delay, got dragged out and about for a bit. https://github.com/CleverRaven/Cataclysm-DDA/pull/13645

**Pull request merged**